conan/tools/cmake/toolchain/toolchain.py

```diff
@@ -214,8 +215,23 @@ def generate(self):
             else:
                 cache_variables[name] = value
 
+        # Get environment information for the presets
+        # FIXME: check if we have an alternative to avoid this way of getting this
+
+        prev_status = self._conanfile.virtualbuildenv
+        build_env = VirtualBuildEnv(self._conanfile).vars()
+        self._conanfile.virtualbuildenv = prev_status
```

Comment:
This pattern is becoming annoying, yes, we should check what other possibilities there could be to avoid this.